test/suite/test_hs21.py

```diff
@@ -0,0 +1,181 @@
+        write_gen = re.search('run_write_gen=\d+', config)
+        self.assertTrue(write_gen is not None)
+        write_gen_str = str()
+        for c in reversed(write_gen.group(0)):
+            if not c.isdigit():
+                self.assertEqual(c, '=')
+                break
+            write_gen_str = c + write_gen_str
+        return int(write_gen_str)
+
+    def test_hs(self):
+        active_files = []
+        value1 = 'a' * 500
+        value2 = 'd' * 500
+
+        # Set up 'numfiles' with 'numrows' entries. We want to create a number of files that
+        # contain active history (content newer than the oldest timestamp).
+        for f in range(self.numfiles):
+            table_uri = 'table:%s.%d' % (self.file_name, f)
+            file_uri = 'file:%s.%d.wt' % (self.file_name, f)
+            # Create a small table.
+            ds = SimpleDataSet(
+                self, table_uri, 0, key_format='S', value_format='S', config='log=(enabled=false)')
+            ds.populate()
+            # Checkpoint to ensure we write the files metadata checkpoint value.
+            self.session.checkpoint()
+            # Get the base write gen of the file so we can compare after the handles get closed.
+            base_write_gen = self.parse_run_write_gen(file_uri)
+            active_files.append((base_write_gen, ds))
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        # Perform a series of updates over our files to check the history store is
+        # working with old and new timestamps.
+        for (_, ds) in active_files:
+            # Load data.
+            self.large_updates(ds.uri, value1, ds, self.nrows // 2 , 1)
+            # Check that all updates are seen.
+            self.check(value1, ds.uri, self.nrows // 2, 1)
+
+            # Load more data with a later timestamp.
+            self.large_updates(ds.uri, value2, ds, self.nrows, 100)
+            # Check that the new updates are only seen after the update timestamp.
+            self.check(value1, ds.uri, self.nrows // 2, 1)
+            self.check(value2, ds.uri, self.nrows, 100)
+
+        # Our sweep scan interval is every 1 second and the amount of idle time needed for a handle to be closed is 3 seconds.
+        # It should take roughly 4 seconds for the sweep server to close our file handles. Lets wait at least double
+        # that to be safe.
+        max = 8
+        sleep = 0
+        # After waiting for the sweep server to remove our idle handles, the only open
+        # handles that should be the metadata file, history store file and lock file.
+        final_numfiles = 3
+        while sleep < max:
+            # We continue doing checkpoints to ensure we sweep the session cache, allowing
+            # idle handles to be removed.
+            self.session.checkpoint()
```

Comment:
I am not sure we need to do that. Is there a reason to believe that checkpoint will sweep the session cache. Is it checkpoint that is doing that or dhandle management as part of checkpoint. Checkpoing also blocks sweep for the time it is running, even though these checkpoints should be very fast.

Reply:
Oh right, the wording of this comment might be confusing. I initially took this approach from a similar usage pattern found in another handle sweeping test [here](https://github.com/wiredtiger/wiredtiger/blob/develop/test/suite/test_sweep01.py#L101-L103) (`test_sweep01.py`)
I might've misunderstood, though I believed it was referring to this operation of the checkpoint process: https://github.com/wiredtiger/wiredtiger/blob/develop/src/txn/txn_ckpt.c#L1078 , eventually removing the dhandle from the session cache through `__session_discard_dhandle`.

I played around with different checkpoint positions, including in the series of earlier updates above. I found checkpointing in this loop to be most consistent with ensuring the handles would eventually be closed.

Reply:
We can change the comment and keep the checkpoint. Checkpoint is giving you the side effect of running session handle sweep. The checkpoint should be fast if the sweep is blocked because of the checkpoint, it should add another second to runtime, which is not the end of the world. I do not expect we will be unlucky often.

Reply:
Sweet. I've updated the comment for this line to better explain the side effect.